### DISCUSSION TITLE
Hitsound lag at 75%-100% playback speed

### DISCUSSION THREAD

**Discussion by 19peytonsmith** [ppy/osu#26133] [Q&A]
### Type

Performance

### Bug description

I know it isn't a timing issue because at 25%-50% it sounds normal, the hitsounds follow according to the beat, but it seems that for the higher playback speeds the hitsounds are a bit delayed. I hope the video properly demonstrates what the issue is, you should be able to hear the delay.

### Screenshots or videos

https://user-images.githubusercontent.com/54728307/190057554-e55ead0e-e4c5-4a84-9148-4164a340b331.mp4

### Version

2022.911.0-lazer

### Logs

[performance.log](https://github.com/ppy/osu/files/9562590/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9562591/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9562592/updater.log)
[database.log](https://github.com/ppy/osu/files/9562593/database.log)
[network.log](https://github.com/ppy/osu/files/9562594/network.log)

**Comment by peppy** [comment]
I brought this up recently, but I don't think I ended up opening an issue for it (may have gotten stuck in my personal TODO list).

The reasoning for this is that `osu-stable` applies an automatic offset when speed is adjusted to make up for inaccuracies in bass timekeeping (or otherwise).

This has not yet been implemented in lazer.

**Comment by peppy** [comment]
Stable code for reference:

![JetBrains Rider 2022-09-14 at 04 13 59](https://user-images.githubusercontent.com/191335/190058000-585e8e51-f789-49e3-b350-4980bf51086b.png)

Of importance, the platform, beatmap and "windows" (aka `usingWasapi` in screenshot) are applied *less* as the playback rate is reduced, and increased as it is increased. This is done in the last line in the screenshot.

When reimplementing, care should be taken to check whether:

- This applies across all OSes
- This applies to frequency adjustments or only "tempo" adjustments

**Comment by peppy** [comment]
To clarify on my own response here: the offset in stable is specific to time stretching. In the editor, we are doing frequency modulation, which *doesn't* require this offset. So this issue is a bit off.

Moving to discussion for now, as I can't personally repro (and the user reports things being *correct* at 25-50% but not 100%, which suggests they need to adjust they UO at 100% instead).
